docs/csharp/language-reference/builtin-types/record.md

```diff
@@ -97,27 +118,25 @@ To implement value equality, the compiler synthesizes the following methods:
 
 If you provide your own implementation of `Equals` in a record type, provide an implementation of `GetHashCode` also.
 
 ## Nondestructive mutation
 
-If you need to mutate immutable properties of a record instance, you can use a `with` expression to achieve *nondestructive mutation*. A `with` expression makes a new record instance that is a copy of an existing record instance, with specified properties and fields modified. You use [object initializer](../../programming-guide/classes-and-structs/object-and-collection-initializers.md) syntax to specify the values to be changed, as shown in the following example:
+If you need to copy an instance with some modifications, you can use a `with` expression to achieve *nondestructive mutation*. A `with` expression makes a new record instance that is a copy of an existing record instance, with specified properties and fields modified. You use [object initializer](../../programming-guide/classes-and-structs/object-and-collection-initializers.md) syntax to specify the values to be changed, as shown in the following example:
 
 :::code language="csharp" source="snippets/shared/RecordType.cs" id="WithExpressions":::
 
 The `with` expression can set positional properties or properties created by using standard property syntax. Non-positional properties must have an `init` or `set` accessor to be changed in a `with` expression.
 
 The result of a `with` expression is a *shallow copy*, which means that for a reference property, only the reference to an instance is copied. Both the original record and the copy end up with a reference to the same instance.
 
-To implement this feature, the compiler synthesizes a clone method and a copy constructor. The virtual clone method returns a new record initialized by the copy constructor. When you use a `with` expression, the compiler creates code that calls the clone method and then sets the properties that are specified in the `with` expression.
+To implement this feature for `record class` types, the compiler synthesizes a clone method and a copy constructor. The virtual clone method returns a new record initialized by the copy constructor. When you use a `with` expression, the compiler creates code that calls the clone method and then sets the properties that are specified in the `with` expression.
 
-If you need different copying behavior, you can write your own copy constructor. If you do that, the compiler won't synthesize one. Make your constructor `private` if the record is `sealed`, otherwise make it `protected`.
+If you need different copying behavior, you can write your own copy constructor. If you do that, the compiler won't synthesize one. Make your constructor `private` if the record is `sealed`, otherwise make it `protected`. You can write a copy constructor for a `record struct`, but that typically isn't needed.
```

Comment:
> You can write a copy constructor for a `record struct`

Is that true? How to do that then?

According to [this example in sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKAPQGYACLUgdSgBcALAUQA8AHAJwFMBnLqAexgAhCLwDGzCGBYAbDgQDeBUsvJlOovmwSkuNNgFdRNUgAU+sGgAoLpABpxSNgJoBKANwElKrJi/LF+CpBpABuEGykLNikALykMBwA7qbmMFYADA7p7n7BYREsmLGR0Ym0dKTydsUArKQAvh6BKvUE9UA==) `record struct` doesn't have even the default copy constructor.

Since, structs don't allow `protected` members, than the custom copy constructor would be `private`. Something [like that](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUBfgMwAERtA6lAgBYCiAHgA7ICmAZwFQA9jABCEYQGMuEMNwA2fCgG8KtTfTr9pI5BloCEaaQloAFEbAQAKG7QAacWg4CaASg1b15Lf9peKAA3CAQ+S2sYOysHbiiED1oQWjYoAVsABhdMrz8AzVVNAF9vEooy+kJCSt8CzVDkQOJaAF5aGD4Ad0ibW2IXYg8Abkr/RsDCNubaLpZWWiLHaYBWWmLR/IL8YgBOW25CEcrS8mKgA==):

```csharp
using System;

public class WithExpressionBasicExample
{
    public record struct Point(int X, int Y)
    {
        private Point(Point point) : this(0, 0)
        {  }
    }

    public static void Main()
    {
        var p1 = new Point(1, 1);
        var p2 = p1 with { X = 5 };
        Console.WriteLine(p2);
    }
}
```

But when I run it, I get `Point { X = 5, Y = 1 }`. So, the zeroing copy constructor is not called.

I have an impression, that a user cannot write a custom copy constructor for `record struct`

Reply:
great points @pkulikov 

You can write a copy constructor, but that won't be called for `with` expressions.  See here:

```
using System;

var pt = new Point(1,2);
var pt2 = new Point(pt);
Console.WriteLine(pt2);
var pt3 = pt2 with {X = 5 };
Console.WriteLine(pt3);

public record struct Point(int X, int Y)
{
    public Point(Point src) : this(src.X, src.Y) {}
}
```

Output:

```
Point { X = 1, Y = 2 }
Point { X = 5, Y = 2 }
```